modules/setting/mailer.go

```diff
@@ -91,6 +90,109 @@ func newMailService() {
 		}
 	}
 
+	if MailService.MailerType == "" {
+		MailService.MailerType = "smtp"
+	}
+
+	// FIXME: DEPRECATED to be removed in v1.18.0
+	deprecatedSetting("mailer", "HOST", "mailer", "SMTP_ADDR")
+	if sec.HasKey("HOST") {
+		givenHost := sec.Key("HOST").String()
+		addr, port, err := net.SplitHostPort(givenHost)
+		if err != nil {
+			log.Fatal("Invalid mailer.HOST (%s): %v", givenHost, err)
+		}
+		MailService.SMTPAddr = addr
+		MailService.SMTPPort = port
+	}
+
+	// FIXME: DEPRECATED to be removed in v1.18.0
+	deprecatedSetting("mailer", "IS_TLS_ENABLED", "mailer", "PROTOCOL")
+	if sec.HasKey("IS_TLS_ENABLED") {
+		if sec.Key("IS_TLS_ENABLED").MustBool() {
+			MailService.Protocol = "smtps"
+		} else {
+			MailService.Protocol = "smtp+startls"
+		}
+	}
+
+	if MailService.SMTPPort == "" {
+		switch MailService.Protocol {
+		case "smtp":
+			MailService.SMTPPort = "25"
+		case "smtps":
+			MailService.SMTPPort = "465"
+		case "smtp+startls":
+			MailService.SMTPPort = "587"
+		}
+	}
+
+	if MailService.MailerType == "smtp" && MailService.Protocol == "" {
+		if strings.ContainsAny(MailService.SMTPAddr, "/\\") {
+			MailService.Protocol = "smtp+unix"
+		} else {
+			switch MailService.SMTPPort {
+			case "25":
+				MailService.Protocol = "smtp"
+			case "465":
+				MailService.Protocol = "smtps"
+			case "587":
+				MailService.Protocol = "smtp+startls"
+			default:
+				log.Fatal("unable to infer unspecified mailer.PROTOCOL from mailer.SMTP_PORT = \"%s\"", MailService.SMTPPort)
+			}
+		}
+	}
+
+	if MailService.Protocol == "smtp" {
+		switch MailService.SMTPAddr {
```

Comment:
Could we use [`net.(IP).IsLoopback()`](https://pkg.go.dev/net#IP.IsLoopback) here?

Reply:
The code is a bit more complicated, but not too complicated; updated

Reply:
Eh, couldn't we use `net.ParseIP(ip).IsLoopback()`?

Reply:
Nope, that'd give a nil error if the address is a hostname, which will probably be most likely

Reply:
Do you think the solution I made is okay? It still won't check if the IPs are *technically* local (e.g. just ones the current host is listening on), but that seems like way more work than it's worth, and it's only a warning anyway.

Reply:
Yes, this seems like the most optimal solution. It handles IP's+hostnames. Just a bummer that with hostnames it will return the IPv4+IPv6 addresses, so the solution is required to return a slice :/